include/openPMD/Datatype.hpp

```diff
@@ -772,6 +771,182 @@ ReturnType switchType( Datatype dt, Action action, Args &&... args )
+        action.OPENPMD_TEMPLATE_OPERATOR() < char >
+        ( std::forward< Args >( args )... ) )
+{
+    using ReturnType = decltype(
+        action.OPENPMD_TEMPLATE_OPERATOR() < char >
+        ( std::forward< Args >( args )... ) );
+    switch( dt )
+    {
+    case Datatype::CHAR:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< char >(
+            std::forward< Args >( args )... );
+    case Datatype::UCHAR:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned char >(
+            std::forward< Args >( args )... );
+    case Datatype::SHORT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< short >(
+            std::forward< Args >( args )... );
+    case Datatype::INT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< int >(
+            std::forward< Args >( args )... );
+    case Datatype::LONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long >(
+            std::forward< Args >( args )... );
+    case Datatype::LONGLONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long long >(
+            std::forward< Args >( args )... );
+    case Datatype::USHORT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned short >(
+            std::forward< Args >( args )... );
+    case Datatype::UINT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned int >(
+            std::forward< Args >( args )... );
+    case Datatype::ULONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned long >(
+            std::forward< Args >( args )... );
+    case Datatype::ULONGLONG:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< unsigned long long >(
+            std::forward< Args >( args )... );
+    case Datatype::FLOAT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< float >(
+            std::forward< Args >( args )... );
+    case Datatype::DOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< double >(
+            std::forward< Args >( args )... );
+    case Datatype::LONG_DOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< long double >(
+            std::forward< Args >( args )... );
+    case Datatype::CFLOAT:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< std::complex< float > >(
+            std::forward< Args >( args )... );
+    case Datatype::CDOUBLE:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< std::complex< double > >(
+            std::forward< Args >( args )... );
+    case Datatype::CLONG_DOUBLE:
+        return action
+            .OPENPMD_TEMPLATE_OPERATOR()< std::complex< long double > >(
+                std::forward< Args >( args )... );
+    case Datatype::STRING:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< std::string >(
+            std::forward< Args >( args )... );
+    case Datatype::BOOL:
+        return action.OPENPMD_TEMPLATE_OPERATOR()< bool >(
+            std::forward< Args >( args )... );
+    case Datatype::DATATYPE:
+        return detail::CallUndefinedDatatype<
+            1000, ReturnType, Action, void, Args &&... >::
```

Comment:
Can we express that magic number in other way?
(Last entry of the data type enum + 1, or a constexpr with a docstring, or so?)

Reply:
I did not yet apply that to the original switchType yet since I'm planning a separate PR for that one.